src/sentry/static/sentry/app/components/forms/form.jsx

```diff
@@ -83,11 +83,8 @@ export default class Form extends React.Component {
   };
 
   onFieldChange = (name, value) => {
-    this.setState({
-      data: {
-        ...this.state.data,
-        [name]: value
-      }
+    this.setState(state => {
```

Comment:
this fixes some propagation bugs, but we could still do the copy inside of here safely if we REALLY wanted identity checks to change

Reply:
This re-renders the entire form on every change

Reply:
yeah thats fine, downstream components could check their values